sage-ios/SAGE/SAGE/SignUpController.swift

```diff
@@ -44,16 +61,105 @@ class SignUpController: UIViewController  {
+    }
+    
+    //
+    // MARK: - Control flow methods that direct sign up flow
+    //
+    
     func createAccountAndProceedToAppWithPicture() {
-        let unverifiedController = UnverifiedViewController()
+        self.showLoadingView()
+        let completion = { (success: Bool) -> Void in
+            // switching back to main thread to ensure that UIKIT methods get called on the main thread
+            dispatch_async(dispatch_get_main_queue(), { () -> Void in
+                if (success) {
+                    let unverifiedController = UnverifiedViewController()
+                    let signUpView = (self.view as! SignUpView)
+                    let photoView = signUpView.photoView
+                    unverifiedController.setImage(photoView.photo.image!)
+                    self.presentViewController(unverifiedController, animated: true, completion: nil)
+                } else {
+                    self.showFailureModal()
+                }
+            })
+        }
+        self.makeAccount(completion)
+    }
+    
+    func makeAccount(completion: ((Bool) -> Void)) {
         let signUpView = (self.view as! SignUpView)
+        let nameView = signUpView.nameView
+        let emailPasswordView = signUpView.emailPasswordView
+        let schoolHoursView = signUpView.schoolHoursView
         let photoView = signUpView.photoView
-        unverifiedController.setImage(photoView.photo.image!)
-        self.presentViewController(unverifiedController, animated: true, completion: nil)
+        
+        let firstName = nameView.firstNameInput.text!
+        let lastName = nameView.lastNameInput.text!
+        let email = emailPasswordView.emailInput.text!
+        let password = emailPasswordView.passwordInput.text!
+        
+        let school = self.schoolDict[schoolHoursView.chooseSchoolButton.titleLabel!.text!] as! Int
+        
+        let hoursString = schoolHoursView.chooseHoursButton.titleLabel!.text!
+        let hours = self.volunteerDict[hoursString] as! Int
+        let verified = false
+        let role = 0
+        
+        var photoData: String
+        if let image = photoView.photo.image {
+            photoData = UIImage.encodedPhotoString(image)
+        } else {
+            let personIcon = FAKIonIcons.personIconWithSize(200)
+            personIcon.setAttributes([NSForegroundColorAttributeName: UIColor(red: 0.4, green: 0.4, blue: 0.4, alpha: 1)])
+            let personImage = personIcon.imageWithSize(CGSizeMake(200, 200))
+            photoData = UIImage.encodedPhotoString(personImage)
+        }
+        
+        LoginHelper.storeUserDataInKeychain(firstName, lastName: lastName, email: email, password: password, school: school, hours: hours, verified: verified)
+        
+        LoginHelper.createUser(firstName, lastName: lastName, email: email, password: password, school: school, hours: hours, role: role, photoData: photoData, completion: completion)
```

Comment:
maybe in createUser, put the storeUserDataInKeyChain inside the method itself (durinc success or so) so that these two lines dont always have to be called together